content/en/blog/2019/monitoring-external-service-traffic/index.md

```diff
@@ -0,0 +1,360 @@
+and operations point of view, it is critical to monitor what external service traffic
+is getting blocked as they might surface possible misconfigurations or a
+security vulnerability if an application is attempting to communicate with a
+service that it should not be allowed to. Similarly, if you currently have a
+policy of allowing any external service access, it is still beneficial to monitor
+the traffic so you can incrementally add explicit Istio configuration to allow
+access and better security your cluster. In either case, having visibility into this
+traffic via telemetry is very helpful as it enables you to create alerts and
+dashboards, and better reason about your security posture. This was a highly
+requested feature by production users of Istio and support for it in release 1.3.
+
+The Istio default metrics are augmented with explicit labels to capture
+blocked and passthrough external service traffic which can be used to filter
+for these cases. This blog will cover how you can use the augmented metrics to
+monitor all external service traffic.
+
+The Istio control plane configures the sidecar proxy with
+predefined clusters called `BlackHoleCluster` and `Passthrough` which block or
+allow all traffic respectively. Let's start with understanding what `BlackHole`
+and `Passthrough` clusters mean in the context of Istio networking.
+
+### What are `BlackHole` and `Passthrough` clusters?
+
+Istio provides users the option to configure how traffic for external services
+is handled by default in the mesh. The Istio control plane by default
+creates and provides configuration for all Kubernetes services to all the sidecars
+in order to preserve the default Kubernetes behavior that all services within the cluster
+are always reachable. This means that unless Istio RBAC or Kubernetes network policies
+are configured, all Kubernetes services continue to communicate with each other seamlessly
+after Istio is added. For services outside the Kubernetes cluster i.e. external
+services, Istio provides two options, first to block all external service access
+(enabled  by setting `global.outboundTrafficPolicy.mode` to `REGISTRY_ONLY`) and
+second to allow all access to external service (enabled  by setting
+`global.outboundTrafficPolicy.mode` to `ALLOW_ANY`). The default option for this
+setting is to allow all external service access. This is where the `BlackHole`
+and `Passthrough` clusters are used.
+
+* **`BlackHoleCluster`** - The `BlackHoleCluster` is a fake virtual cluster created
+  in the Envoy configuration when `global.outboundTrafficPolicy.mode` is set to
+  `REGISTRY_ONLY`. In this mode, all traffic to external service is blocked unless
+  [`ServiceEntries`](/docs/reference/config/networking/v1alpha3/service-entry)
+  are explicitly added for each service. To implement this, the default virtual
+  outbound listener at `0.0.0.0:15001` which uses `SO_ORIGINAL_DST` is setup
```

Comment:
maybe add a link to Envoy docs on original destination cluster: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery#original-destination ?

Reply:
do you know how to get a stable link for these? I feel like Envoy might change thinks on us and the links will get outdated.

Reply:
Yes, just use those links as-is. Once in a while, Envoy rejiggers things and we just go through our site and patch up the links.